main/views.py

```diff
@@ -129,28 +134,48 @@ def edit_profile(request):
 @auth_required
 def get_profile(request):
     user = request.user
-    user_profile = Profile.objects.get(user=user)
-    attendance = Attendance.objects.filter(user=user).order_by("-check_in")[0:7]
-    attendance = [attendance.to_dict() for attendance in attendance]
-    print(attendance)
-    graph_data = {"labels": [], "data": []}
-    hours_worked = 0
-    avg_hours = 0
-    days_worked = 0
-    for attendance in attendance:
-        print(attendance["check_in"], attendance["check_out"])
-        graph_data["labels"].append(attendance["check_in"].date())
-        try:
-            graph_data["data"].append(int(attendance["check_out"].hour - attendance["check_in"].hour))
-            hours_worked += int(attendance["check_out"].hour - attendance["check_in"].hour)
-        except:
-            graph_data["data"].append(0)
-        print(graph_data)
-        avg_hours = hours_worked / len(graph_data["data"])
-        days_worked = len(graph_data["data"])
-    return JsonResponse({"status": 200, "data": user_profile.to_dict(), "graph_data": graph_data,
-                         "stats_data": {"hours_worked": hours_worked, "avg_hours": avg_hours,
-                                        "days_worked": days_worked}})
+    try:
+        user_profile = Profile.objects.get(user=user)
+
+        attendance = Attendance.objects.filter(user=user).order_by("-check_in")[:7]
+        attendance = [att.to_dict() for att in attendance]
+
+        graph_data = {"labels": [], "data": []}
+        hours_worked = 0
+        avg_hours = 0
+        days_worked = 0
+
+        for att in attendance:
+            graph_data["labels"].append(att["check_in"].date())
+            # Check if check_out exists
+            if att["check_out"] is not None:
+                worked_hours = att["check_out"].hour - att["check_in"].hour
+                graph_data["data"].append(worked_hours)
```

Comment:
please revert it back, chatgpt doesn't know how the data is populated in our front end :slightly_smiling_face:

Reply:
ok